src/autoscaler/eventgenerator/config/config.go

```diff
@@ -29,102 +37,181 @@ const (
-	ScalingEngine             ScalingEngineConfig   `yaml:"scalingEngine"`
-	MetricCollector           MetricCollectorConfig `yaml:"metricCollector"`
-	DefaultStatWindowSecs     int                   `yaml:"defaultStatWindowSecs"`
-	DefaultBreachDurationSecs int                   `yaml:"defaultBreachDurationSecs"`
-	CircuitBreaker            CircuitBreakerConfig  `yaml:"circuitBreaker"`
-	HttpClientTimeout         time.Duration         `yaml:"http_client_timeout"`
+	Logging                   helpers.LoggingConfig        `yaml:"logging" json:"logging"`
+	Server                    helpers.ServerConfig         `yaml:"server" json:"server"`
+	CFServer                  helpers.ServerConfig         `yaml:"cf_server" json:"cf_server"`
+	Pool                      *PoolConfig                  `yaml:"pool" json:"pool"`
+	Health                    helpers.HealthConfig         `yaml:"health" json:"health"`
+	Db                        map[string]db.DatabaseConfig `yaml:"db"`
+	Aggregator                *AggregatorConfig            `yaml:"aggregator" json:"aggregator,omitempty"`
+	Evaluator                 *EvaluatorConfig             `yaml:"evaluator" json:"evaluator,omitempty"`
+	ScalingEngine             ScalingEngineConfig          `yaml:"scalingEngine" json:"scalingEngine"`
+	MetricCollector           MetricCollectorConfig        `yaml:"metricCollector" json:"metricCollector"`
+	DefaultStatWindowSecs     int                          `yaml:"defaultStatWindowSecs" json:"defaultStatWindowSecs"`
+	DefaultBreachDurationSecs int                          `yaml:"defaultBreachDurationSecs" json:"defaultBreachDurationSecs"`
+	CircuitBreaker            *CircuitBreakerConfig        `yaml:"circuitBreaker,omitempty" json:"circuitBreaker,omitempty"`
+	HttpClientTimeout         *time.Duration               `yaml:"http_client_timeout,omitempty" json:"http_client_timeout,omitempty"`
 }
 
-func LoadConfig(config []byte) (*Config, error) {
+func LoadConfig(filepath string, vcapReader configutil.VCAPConfigurationReader) (*Config, error) {
 	conf := defaultConfig()
-	dec := yaml.NewDecoder(bytes.NewBuffer(config))
+
+	if err := loadYamlFile(filepath, &conf); err != nil {
+		return nil, err
+	}
+
+	if err := loadVcapConfig(&conf, vcapReader); err != nil {
+		return nil, err
+	}
+
+	return &conf, nil
+}
+
+func loadEventgeneratorConfig(conf *Config, vcapReader configutil.VCAPConfigurationReader) error {
+	data, err := vcapReader.GetServiceCredentialContent("eventgenerator-config", "eventgenerator-config")
+	if err != nil {
+		return fmt.Errorf("%w: %v", ErrEventgeneratorConfigNotFound, err)
+	}
+	return yaml.Unmarshal(data, conf)
+}
+
+func loadVcapConfig(conf *Config, vcapReader configutil.VCAPConfigurationReader) error {
+	if !vcapReader.IsRunningOnCF() {
+		return nil
+	}
+	// enable plain text logging. See src/autoscaler/helpers/logger.go
+	conf.Logging.PlainTextSink = true
+
+	conf.CFServer.Port = vcapReader.GetPort()
+	conf.Server.Port = 0
```

Comment:
May be add comment:

"
Make sure that the bosh server and cf app does not have the same port (as it may happen). Hence, setting the bosh server port to 0
"

Reply:
added in dd6993f89